### PR TITLE
[MM-54798] Convert `./components/widgets/separator/basic-separator.tsx` to function component

### PR DESCRIPTION
**Pull request by fyfirman**
#### Summary
Convert `./components/widgets/separator/basic-separator.tsx` to function component.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/24762
Jira [MM-54798](https://mattermost.atlassian.net/browse/MM-54798)

#### Release Note

```release-note
NONE
```

[MM-54798]: https://mattermost.atlassian.net/browse/MM-54798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/channels/src/components/widgets/separator/basic-separator.tsx [hunk 1]
```diff
@@ -9,21 +9,20 @@ type Props = {
...
-                <hr className='separator__hr'/>
-                {children && (
-                    <div className='separator__text'>
-                        {children}
-                    </div>
-                )}
-            </div>
-        );
-    }
+function BasicSeparator({children}: Props) {
```
Comment (larkox) [inline webapp/channels/src/components/widgets/separator/basic-separator.tsx:12]: Can you make this a arrow function? I think there is not much difference, but I want to keep as much as possible the coherence between different parts of the code, and that means all function components to be arrow functions :sweat_smile:

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just one small nitpick about style.

File: webapp/channels/src/components/widgets/separator/basic-separator.tsx (hunk 1 above)
Comment (fyfirman) [reply webapp/channels/src/components/widgets/separator/basic-separator.tsx]: fixed with [f6202a0](https://github.com/mattermost/mattermost/pull/24788/commits/f6202a0d91cf59038bbbd01a47272cfda7b20a99)

**Review by larkox — APPROVED** [review]

**Comment by fyfirman** [comment]
It there anything I have to do with the failed CI? @larkox
I don't really understand with the error though.

**Comment by larkox** [comment]
@fyfirman don't worry that particular error is on our side :smile:

**Review by hmhealey — APPROVED** [review]
Thanks for the PR and for the help with this migration, @fyfirman. This looks good to me

**Comment by hmhealey** [comment]
Skipping QA review since this component doesn't have any functionality to test

**Pull request merged**